src/main/java/com/cburch/logisim/LogisimVersion.java

```diff
@@ -9,61 +9,113 @@
   }
 
   public LogisimVersion(int major, int minor, int patch, String suffix) {
-    this.major = major;
-    this.minor = minor;
-    this.patch = patch;
-    if (suffix == null) {
-      suffix = "";
-    }
-    this.suffix = suffix.strip();
+    suffix = (suffix == null) ? "" : suffix;
+    final var versionString = String.format("%d.%d.%d%s", major, minor, patch, suffix);
+    initFromVersionString(versionString);
   }
 
   /**
    * Parse a string containing a version number and returns the corresponding LogisimVersion object.
    * No exception is thrown if the version string contains non-integers, because literal values are
    * allowed.
    *
-   * <p>Supported version string formats are `X.Y.Z` or `X.Y.Z-SUFFIX`
+   * <p>Supported version string formats are `X.Y.Z`, `X.Y.Z-suffix` or `X.Y.Zsuffix`.
    *
    * @return LogisimVersion built from the string passed as parameter
    */
   public static LogisimVersion fromString(String versionString) {
+    return new LogisimVersion().initFromVersionString(versionString);
+  }
+
+//  private static final String fromStringErrorMsg = "Version segments must be non-negative integers, '%s' found.";
+
+  private LogisimVersion initFromVersionString(String versionString) throws IllegalArgumentException {
     var major = 0;
     var minor = 0;
     var patch = 0;
+    var separator = "";
     var suffix = "";
 
-    // Let's see if we have suffix segment or not.
-    final var segments = versionString.split("-");
-
-    if (segments.length > 0) {
-      if (segments.length == 2) {
-        suffix = segments[1].strip();
-      }
+    var pattern = "^(\\d+.\\d+.\\d+)(.*)$";
+    var m = Pattern.compile(pattern, Pattern.CASE_INSENSITIVE).matcher(versionString);
+    if (m.matches()) {
+      final var verStr = m.group(1);
+      final var sufStr = m.group(2);
 
-      final var parts = segments[0].split("\\.");
+      final var parts = m.group(1).split("\\.");
       try {
         if (parts.length >= 1) major = Integer.parseInt(parts[0]);
         if (parts.length >= 2) minor = Integer.parseInt(parts[1]);
         if (parts.length >= 3) patch = Integer.parseInt(parts[2]);
-      } catch (NumberFormatException ignored) {
-        // Just ignore. We will just fall back to `0`
+      } catch (NumberFormatException ex) {
+        throw new IllegalArgumentException(String.format("Version segments must be non-negative integers, '%s' found.", verStr));
+      }
+
+      // suffix part
+      if (sufStr != null) {
+        if(sufStr.length() == 1) {
+          pattern = "^[a-z]+$";
+          m = Pattern.compile(pattern, Pattern.CASE_INSENSITIVE).matcher(sufStr);
+          if (!m.matches()) {
+            throw new IllegalArgumentException(String.format("Suffix must start with a letter, '%s' found.", sufStr));
+          }
+          suffix = sufStr;
+        } else if(sufStr.length() > 1) {
+          // for single character suffix it can only be letter
+          pattern = "^(-)?([a-z]+([a-z\\d])*)+$";
```

Comment:
Could you explain why `"^(-)?([a-z][a-z\\d]*)$" doesn't work here? I admit I haven't use Java for regex matching so I am not very familiar with the format. I'm asking so I understand.

Reply:
It would just fine. It's basically simpler version so I took it. Thanks.